### PR TITLE
Update to JupyterLab 4

### PR DESCRIPTION
**Pull request by andrewfulton9**
This PR upgrades pyviz_comms to be a jupyterlab4 extension. I've tested it by ensuring it builds successfully, the tests pass, and that I can run holoviz with it. Are there other tests I should run?

A couple of the things I am unsure about:
- How to test the conda recipe
- If the new workflow files generated should be included (I have done so, but I can remove them if need be)
- Whether the following files should be removed:
    - dodo.py
    - install.json

Thanks!

### CODE REVIEW

**Comment by philippjfr** [comment]
Wow, thank you @andrewfulton9, I owe you a beer or other beverage of your choice! I'll have a go at testing this tomorrow.

**Comment by philippjfr** [comment]
I'm fine with removing dodo.py and install.json. We're moving away from `doit` anyway.

**Comment by philippjfr** [comment]
Might also work on setting up some integration tests with Playwright that test Panel, HoloViews and the Panel Preview button work both in classic notebook and JupyterLab.

**Comment by andrewfulton9** [comment]
I'm still trying to get all the checks passing. I'm at a bit of a loss for how to get the `Check Release` check to pass.

**Comment by philippjfr** [comment]
I can hopefully take a look soon, but likely not today. This seems to be pretty close, so already a huge thanks for all your efforts.

**Comment by andrewfulton9** [comment]
> I can hopefully take a look soon, but likely not today. This seems to be pretty close, so already a huge thanks for all your efforts.

No problem, thank you!

**Comment by krassowski** [comment]
>  I'm at a bit of a loss for how to get the Check Release check to pass.

@andrewfulton9 I run into the same issue and ended up adding `steps_to_skip: check-npm` for now to silence it (https://github.com/jupyterlab-contrib/jupyterlab-vim/pull/85/commits/dfe8101f33ba79ff374471fcb8afb537ee67dace). Tracking it in https://github.com/jupyterlab/extension-template/issues/36.

File: package.json [hunk 1]
```diff
@@ -26,40 +29,48 @@
...
-    "eslint": "eslint . --ext .ts,.tsx --fix",
-    "eslint:check": "eslint . --ext .ts,.tsx",
-    "install:extension": "jupyter labextension develop --overwrite .",
+    "clean:lintcache": "rimraf .eslintcache .stylelintcache",
+    "eslint": "jlpm eslint:check --fix",
+    "eslint:check": "eslint . --cache --ext .ts,.tsx",
+    "install:extension": "jlpm build",
+    "lint": "jlpm prettier && jlpm eslint",
+    "lint:check": "jlpm prettier:check && jlpm eslint:check",
     "prepare": "jlpm run clean && jlpm run build:prod",
```
Comment (krassowski) [inline package.json:48]: As per https://github.com/jupyterlab/extension-template/issues/36#issuecomment-1657147331 this is the proper fix for `npm-check`:

```suggestion
```

**Comment by andrewfulton9** [comment]
@philippjfr, have you gotten a chance to take a look at this?

**Comment by philippjfr** [comment]
I had a brief look and then I suddenly had to organize a move of our apartment. Will be back to work Monday.

**Comment by philippjfr** [comment]
I'll merge despite the conda package release not being handled. I think we should move away from the pyviz channel in any case as we now have much more control over both defaults and conda-forge channel releases. If there is a strong need for it I may decide to restore it later, which is why I've not removed the recipe yet but that shouldn't hold up merge and releases.

**Pull request merged**